genologics/lims.py

```diff
@@ -379,21 +380,24 @@ def _get_instances(self, klass, params=dict()):
             root = self.get(node.attrib['uri'], params=params)
         return result
 
-    def get_batch(self, instances):
-        "Get the content of a set of instances using the efficient batch call."
+    def get_batch(self, instances, force=False):
+        """Get the content of a set of instances using the efficient batch call."""
         if not instances:
             return []
-        klass = instances[0].__class__
+
         root = ElementTree.Element(nsmap('ri:links'))
+        klass = None
         result = []
         needs_request=False
         for instance in instances:
-            try:
-                result.append(self.cache[instance.uri])
-            except:
-                needs_request=True
+            if not klass:
```

Comment:
The diff is a bit messed up.. This test and the line below it replaces "klass = instances[0].__class__". I moved the assignment into the loop to support any iterable in the function argument. The wasted CPU is even worse than if it was an if vs. exception, but i think it's still worthwhile to support things like generator expressions in the argument. 

The exception in self.cache[instance.uri] is replaced with "instance.root is None". I believe the original version will never throw an exception here, since the __init__ function in Entity will add *any* new entity object to the cache, as soon as it's constructed (and the custom __new__ will return objects from the cache instead of new duplicate objects).